### PR TITLE
Slider uses visitor from pattern-validator

### PR DESCRIPTION
**Pull request by rpiazza**
make slider work with base visitor from pattern-validator

### CODE REVIEW

File: stix2slider/convert_stix.py [hunk 1]
```diff
@@ -669,123 +692,198 @@ def convert_report(r20):
...
-        warn("labels not representable in a STIX 1.x ToolInformation.  Found in %s", 502, tool20["id"])
+    if "name" in tool2x:
+        tool1x.title = tool2x["name"]
+    if "description" in tool2x:
+        tool1x.description = tool2x["description"]
+    if "tool_version" in tool2x:
+        tool1x.version = tool2x["tool_version"]
+    if "labels" in tool2x:
+        warn("labels not representable in a STIX 1.x ToolInformation.  Found in %s", 502, tool2x["id"])
         # bug in python_stix prevents using next line of code
```
Comment (clenk) [inline stix2slider/convert_stix.py:767]: Is this bug in the python_stix issue tracker? Should we prioritize fixing it?

File: stix2slider/convert_stix.py [hunk 2]
```diff
@@ -543,113 +549,130 @@ def convert_identity(ident20):
...
-        malware1x.add_description(malware20["description"])
-    types = convert_open_vocabs_to_controlled_vocabs(malware20["labels"], MALWARE_LABELS_MAP)
+    if "name" in malware2x:
+        malware1x.add_name(malware2x["name"])
+    if "description" in malware2x:
+        malware1x.add_description(malware2x["description"])
+    if VERSION_OF_STIX_2x == "2.0":
+        types = convert_open_vocabs_to_controlled_vocabs(malware2x["labels"], MALWARE_LABELS_MAP)
+    else:
+        types = convert_open_vocabs_to_controlled_vocabs(malware2x["malware_type"], MALWARE_LABELS_MAP)
```
Comment (clenk) [inline stix2slider/convert_stix.py]: Should be `malware_types`, plural.

File: stix2slider/convert_stix.py [hunk 3]
```diff
@@ -669,123 +692,198 @@ def convert_report(r20):
...
+    ciq_object.specification = STIXCIQIdentity3_0()
+    ciq_object.specification.party_name = identity_object.name
+    return ciq_object
+
+
+def add_location_to_information_source(source_1x_obj, target_obj):
+    if isinstance(source_1x_obj, ThreatActor):
+        if source_1x_obj.information_source == None:
+            source_1x_obj.information_source = InformationSource()
+        source_1x_obj.information_source
```
Comment (clenk) [inline stix2slider/convert_stix.py]: Is there something missing from this line, or is it unnecessary?

File: stix2slider/test/test_idioms.py [hunk 4]
```diff
@@ -137,7 +137,8 @@ def test_idiom_mapping(test_file, stored_master):
 
 
 def pytest_generate_tests(metafunc):
-    setup_tests()
+    version = os.environ['VERSION']
```
Comment (clenk) [inline stix2slider/test/test_idioms.py:140]: Maybe we should use a more specific name for the environment variable. Also, we should configure Tox to set this variable with both versions to make sure we test everything.

**Review by clenk — CHANGES_REQUESTED** [review]
Looking good. Just a few comments to look at, and there's some style issues to fix (I re-ran Travis since we made a new pattern-validator release).

File: stix2slider/convert_stix.py (hunk 2 above)
Comment (rpiazza) [reply stix2slider/convert_stix.py]: Good catch

File: stix2slider/convert_stix.py (hunk 1 above)
Comment (rpiazza) [reply stix2slider/convert_stix.py:767]: see [python cybox tracker #310](https://github.com/CybOXProject/python-cybox/issues/310).  The fix apparently didn't make it back to the slider.

Comment (rpiazza) [reply stix2slider/convert_stix.py:767]: see #39

**Review by clenk — APPROVED** [review]

**Comment by rpiazza** [comment]
@clenk 

If you review latest changes and merge this - I can do the release.

**Review by clenk — APPROVED** [review]
Looks good, thanks @rpiazza!

**Pull request merged**